pkg/publish/default.go

```diff
@@ -40,8 +40,9 @@ type defalt struct {
 	base      string
 	t         http.RoundTripper
 	userAgent string
-	auth      authn.Authenticator
 	namer     Namer
+	auth      authn.Authenticator
```

Comment:
This shouldn't have an `auth` field anymore, only `keychain`. `publish.WithAuth` should set `keychain = staticKeychain{auth}`.